File: frontend/src/services/followService.js
import api from '../utils/api';

export const followService = {
    followUser: async (userId) => {
        const response = await api.post(`/users/${userId}/follow`);
        return response.data;
    },

    unfollowUser: async (userId) => {
        const response = await api.delete(`/users/${userId}/follow`);
        return response.data;
    },

    isFollowing: async (userId) => {
        const response = await api.get(`/users/${userId}/is-following`);
        return response.data;
    },

    getFollowerCount: async (userId) => {
        const response = await api.get(`/users/${userId}/followers/count`);
        return response.data;
    },

    getFollowingCount: async (userId) => {
        const response = await api.get(`/users/${userId}/following/count`);
        return response.data;
    },

    getFollowers: async (userId, page = 0, size = 10) => {
        const response = await api.get(`/users/${userId}/followers?page=${page}&size=${size}`);
        return response.data;
    },

    getFollowing: async (userId, page = 0, size = 10) => {
        const response = await api.get(`/users/${userId}/following?page=${page}&size=${size}`);
        return response.data;
    }
};

File: src/main/java/com/Blog/Platform/User/Service/FollowService.java
package com.Blog.Platform.User.Service;

import com.Blog.Platform.User.DTO.PublicUserProfileResponse;
import com.Blog.Platform.User.Model.Follow;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface FollowService {
    void followUser(UUID targetUserId);

    void unfollowUser(UUID targetUserId);

    boolean isFollowing(UUID targetUserId);

    long getFollowerCount(UUID userId);

    long getFollowingCount(UUID userId);

    Page<PublicUserProfileResponse> getFollowers(UUID userId, Pageable pageable);

    Page<PublicUserProfileResponse> getFollowing(UUID userId, Pageable pageable);

    java.util.List<com.Blog.Platform.User.Model.User> getFollowingUsers(UUID userId);

    java.util.List<com.Blog.Platform.User.Model.User> getFollowersList(UUID userId);
}
